crates/api_common/src/utils.rs

```diff
@@ -516,6 +518,59 @@ pub fn local_site_opt_to_sensitive(local_site: &Option<LocalSite>) -> bool {
     .unwrap_or(false)
 }
 
+async fn get_url_blocklist() -> &'static RwLock<Option<RegexSet>> {
```

Comment:
I'm probably alone here, but I really don't think this caching is worth the effort. We aren't doing this for the slur regex, and that's run in the same circumstances.

If someone wants to add caching for this later, they always can, like the local site cache for the federation workers.

Reply:
You're not alone here bro. For this PR, I'm fine with anything that works correctly with horizontal scaling.

Reply:
The docs explicitly mention that [constructing a regex is relatively expensive](https://docs.rs/regex/latest/regex/struct.Regex.html#method.new). Without a cache the regex may be constructed hundreds of times per second to scan all the posts, so it could easily affect performance.